### ISSUE TITLE
first_api container is unable to connect to mysql

### ISSUE THREAD

**Issue by p-state** [vrtadmin/FIRST-server#28]
After the fresh installation of the FIRST server using Docker, I can not use it. Logs are provided below.

docker logs first_api_1
```
Using existing SSL certificate
[1st] IOError: [Errno 2] No such file or directory: '/home/first/first/../first_config.json'
Traceback (most recent call last):
  File "/usr/local/lib/python3.6/dist-packages/django/db/backends/base/base.py", line 216, in ensure_connection
    self.connect()
  File "/usr/local/lib/python3.6/dist-packages/django/db/backends/base/base.py", line 194, in connect
    self.connection = self.get_new_connection(conn_params)
  File "/usr/local/lib/python3.6/dist-packages/django/db/backends/mysql/base.py", line 227, in get_new_connection
    return Database.connect(**conn_params)
  File "/usr/local/lib/python3.6/dist-packages/MySQLdb/__init__.py", line 84, in Connect
    return Connection(*args, **kwargs)
  File "/usr/local/lib/python3.6/dist-packages/MySQLdb/connections.py", line 164, in __init__
    super(Connection, self).__init__(*args, **kwargs2)
MySQLdb._exceptions.OperationalError: (2002, "Can't connect to local MySQL server through socket '/var/run/mysqld/mysqld.sock' (2)")

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/home/first/manage.py", line 22, in <module>
    execute_from_command_line(sys.argv)
  File "/usr/local/lib/python3.6/dist-packages/django/core/management/__init__.py", line 381, in execute_from_command_line
    utility.execute()
  File "/usr/local/lib/python3.6/dist-packages/django/core/management/__init__.py", line 375, in execute
    self.fetch_command(subcommand).run_from_argv(self.argv)
  File "/usr/local/lib/python3.6/dist-packages/django/core/management/base.py", line 316, in run_from_argv
    self.execute(*args, **cmd_options)
  File "/usr/local/lib/python3.6/dist-packages/django/core/management/base.py", line 350, in execute
    self.check()
  File "/usr/local/lib/python3.6/dist-packages/django/core/management/base.py", line 379, in check
    include_deployment_checks=include_deployment_checks,
  File "/usr/local/lib/python3.6/dist-packages/django/core/management/commands/migrate.py", line 59, in _run_checks
    issues = run_checks(tags=[Tags.database])
  File "/usr/local/lib/python3.6/dist-packages/django/core/checks/registry.py", line 71, in run_checks
    new_errors = check(app_configs=app_configs)
  File "/usr/local/lib/python3.6/dist-packages/django/core/checks/database.py", line 10, in check_database_backends
    issues.extend(conn.validation.check(**kwargs))
  File "/usr/local/lib/python3.6/dist-packages/django/db/backends/mysql/validation.py", line 9, in check
    issues.extend(self._check_sql_mode(**kwargs))
  File "/usr/local/lib/python3.6/dist-packages/django/db/backends/mysql/validation.py", line 13, in _check_sql_mode
    with self.connection.cursor() as cursor:
  File "/usr/local/lib/python3.6/dist-packages/django/db/backends/base/base.py", line 255, in cursor
    return self._cursor()
  File "/usr/local/lib/python3.6/dist-packages/django/db/backends/base/base.py", line 232, in _cursor
    self.ensure_connection()
  File "/usr/local/lib/python3.6/dist-packages/django/db/backends/base/base.py", line 216, in ensure_connection
    self.connect()
  File "/usr/local/lib/python3.6/dist-packages/django/db/utils.py", line 89, in __exit__
    raise dj_exc_value.with_traceback(traceback) from exc_value
  File "/usr/local/lib/python3.6/dist-packages/django/db/backends/base/base.py", line 216, in ensure_connection
    self.connect()
  File "/usr/local/lib/python3.6/dist-packages/django/db/backends/base/base.py", line 194, in connect
    self.connection = self.get_new_connection(conn_params)
  File "/usr/local/lib/python3.6/dist-packages/django/db/backends/mysql/base.py", line 227, in get_new_connection
    return Database.connect(**conn_params)
  File "/usr/local/lib/python3.6/dist-packages/MySQLdb/__init__.py", line 84, in Connect
    return Connection(*args, **kwargs)
  File "/usr/local/lib/python3.6/dist-packages/MySQLdb/connections.py", line 164, in __init__
    super(Connection, self).__init__(*args, **kwargs2)
django.db.utils.OperationalError: (2002, "Can't connect to local MySQL server through socket '/var/run/mysqld/mysqld.sock' (2)")
AH00558: apache2: Could not reliably determine the server's fully qualified domain name, using 172.18.0.4. Set the 'ServerName' directive globally to suppress this message
```

docker logs first_mysql_1
```
2019-03-12T13:01:24.860546Z 0 [Warning] TIMESTAMP with implicit DEFAULT value is deprecated. Please use --explicit_defaults_for_timestamp server option (see documentation for more details).
2019-03-12T13:01:24.878578Z 0 [Note] mysqld (mysqld 5.7.25) starting as process 1 ...
2019-03-12T13:01:24.932836Z 0 [Note] InnoDB: PUNCH HOLE support available
2019-03-12T13:01:24.932898Z 0 [Note] InnoDB: Mutexes and rw_locks use GCC atomic builtins
2019-03-12T13:01:24.932904Z 0 [Note] InnoDB: Uses event mutexes
2019-03-12T13:01:24.932908Z 0 [Note] InnoDB: GCC builtin __atomic_thread_fence() is used for memory barrier
2019-03-12T13:01:24.932912Z 0 [Note] InnoDB: Compressed tables use zlib 1.2.11
2019-03-12T13:01:24.932916Z 0 [Note] InnoDB: Using Linux native AIO
2019-03-12T13:01:24.938259Z 0 [Note] InnoDB: Number of pools: 1
2019-03-12T13:01:24.960804Z 0 [Note] InnoDB: Using CPU crc32 instructions
2019-03-12T13:01:24.988673Z 0 [Note] InnoDB: Initializing buffer pool, total size = 128M, instances = 1, chunk size = 128M
2019-03-12T13:01:25.160340Z 0 [Note] InnoDB: Completed initialization of buffer pool
2019-03-12T13:01:25.169485Z 0 [Note] InnoDB: If the mysqld execution user is authorized, page cleaner thread priority can be changed. See the man page of setpriority().
2019-03-12T13:01:25.206608Z 0 [Note] InnoDB: Highest supported file format is Barracuda.
2019-03-12T13:01:25.315532Z 0 [Note] InnoDB: Creating shared tablespace for temporary tables
2019-03-12T13:01:25.315713Z 0 [Note] InnoDB: Setting file './ibtmp1' size to 12 MB. Physically writing the file full; Please wait ...
2019-03-12T13:01:25.451939Z 0 [Note] InnoDB: File './ibtmp1' size is now 12 MB.
2019-03-12T13:01:25.467404Z 0 [Note] InnoDB: 96 redo rollback segment(s) found. 96 redo rollback segment(s) are active.
2019-03-12T13:01:25.467461Z 0 [Note] InnoDB: 32 non-redo rollback segment(s) are active.
2019-03-12T13:01:25.468061Z 0 [Note] InnoDB: Waiting for purge to start
2019-03-12T13:01:25.520594Z 0 [Note] InnoDB: 5.7.25 started; log sequence number 12358972
2019-03-12T13:01:25.521210Z 0 [Note] InnoDB: Loading buffer pool(s) from /var/lib/mysql/ib_buffer_pool
2019-03-12T13:01:25.522360Z 0 [Note] Plugin 'FEDERATED' is disabled.
2019-03-12T13:01:25.721923Z 0 [Note] Found ca.pem, server-cert.pem and server-key.pem in data directory. Trying to enable SSL support using them.
2019-03-12T13:01:25.724115Z 0 [Warning] CA certificate ca.pem is self signed.
2019-03-12T13:01:25.725024Z 0 [Note] InnoDB: Buffer pool(s) load completed at 190312 13:01:25
2019-03-12T13:01:25.730797Z 0 [Note] Server hostname (bind-address): '*'; port: 3306
2019-03-12T13:01:25.747044Z 0 [Note] IPv6 is available.
2019-03-12T13:01:25.767023Z 0 [Note]   - '::' resolves to '::';
2019-03-12T13:01:25.767138Z 0 [Note] Server socket created on IP: '::'.
2019-03-12T13:01:25.795319Z 0 [Warning] Insecure configuration for --pid-file: Location '/var/run/mysqld' in the path is accessible to all OS users. Consider choosing a different directory.
2019-03-12T13:01:25.836635Z 0 [Warning] 'user' entry 'root@localhost' ignored in --skip-name-resolve mode.
2019-03-12T13:01:25.836739Z 0 [Warning] 'user' entry 'mysql.session@localhost' ignored in --skip-name-resolve mode.
2019-03-12T13:01:25.836752Z 0 [Warning] 'user' entry 'mysql.sys@localhost' ignored in --skip-name-resolve mode.
2019-03-12T13:01:25.837143Z 0 [Warning] 'db' entry 'performance_schema mysql.session@localhost' ignored in --skip-name-resolve mode.
2019-03-12T13:01:25.837160Z 0 [Warning] 'db' entry 'sys mysql.sys@localhost' ignored in --skip-name-resolve mode.
2019-03-12T13:01:25.854110Z 0 [Warning] 'proxies_priv' entry '@ root@localhost' ignored in --skip-name-resolve mode.
2019-03-12T13:01:25.897319Z 0 [Warning] 'tables_priv' entry 'user mysql.session@localhost' ignored in --skip-name-resolve mode.
2019-03-12T13:01:25.897370Z 0 [Warning] 'tables_priv' entry 'sys_config mysql.sys@localhost' ignored in --skip-name-resolve mode.
2019-03-12T13:01:26.189657Z 0 [Note] Event Scheduler: Loaded 0 events
2019-03-12T13:01:26.190141Z 0 [Note] mysqld: ready for connections.
Version: '5.7.25'  socket: '/var/run/mysqld/mysqld.sock'  port: 3306  MySQL Community Server (GPL)
```

Perhaps the problem is that `first_config.json` is missing, but I cannot find documentation on how to properly configure this file.

**Comment by p-state** [comment]
I've tried to use the following `first_config.json`, but the same error appears.
```
{
    "secret_key" : "top secret",

    "db_engine" : "django.db.backends.mysql",
    "db_dbname" : "first_db",
    "db_user" : "root",
    "db_password" : "password12345",
    "db_host" : "localhost",
    "db_port" : 3306,

    "debug" : true,
    "allowed_hosts" : ["localhost", "here is ip of my server"],

    "oauth_path" : "/usr/local/etc/google_secret.json"
}
```

**Comment by p-state** [comment]
I managed to solve this issue by assigning “db_host” to the first_mysql's IP address. The IP address can be obtained using `docker inspect first_mysql_1 | grep IPAddress`. After updating `/home/first/first_config.json`, I needed to restart the apache2 service.

I think this process should be explained somewhere. 

Also, you have an redundant comma at the end of example_config.json, which breaks JSON parser.
